SwiftyJSONAPI/JSONAPIResource.swift

```diff
@@ -102,13 +105,40 @@ public class JSONAPIResource: JSONPrinter {
                 
                 resource.attributes    = includedResource.attributes
                 resource.relationships = includedResource.relationships
+            
                 
-                if !resource.relationships.isEmpty {
+                if !resource.relationships.isEmpty, resource.shouldLoadResource(withCachedResources: &cachedResources) {
+                    
                     resource.parent = self.parent
-                    resource.loadResources(withIncludedResources: includedResources)
+                    resource.loadResources(withIncludedResources: includedResources, cachedResources: &cachedResources)
                 }
+                
                 resource.loaded = .Loaded
             }
         }
     }
 }
+
+private extension JSONAPIResource {
+    
+    // Checks if a resource has been loaded already, prevents bidirectional relationships from being recursively called
+    func shouldLoadResource(withCachedResources cachedResources: inout CachedResources) -> Bool {
+        
+        if var cachedResourceIds = cachedResources[type] {
+            
+            if cachedResourceIds.contains(id) {
+                
+                return false
+            } else {
+                
+                cachedResourceIds.insert(id)
+                cachedResources[type] = cachedResourceIds
+                return true
+            }
+        } else {
+            
+            cachedResources[type] = Set([id])
```

Comment:
is compiler not happy with just = [id] ? it should know it's a set from cachedResources type no?